src/compiler/parser.ts

```diff
@@ -1891,6 +1891,10 @@ namespace Parser {
                     }
 
                     if (pos >= 0) {
+                        // skip all statements in range already completely consumed by the reparsed statement
+                        while (sourceFile.statements[pos].end < statement.end) {

```

Comment:
I missed this before, but this no longer checks that `pos < sourceFile.statements.length`, but it still should